components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java

```diff
@@ -817,6 +859,56 @@ public String publishModel(int tenantId, String userName, long modelId) throws I
         return lines;
     }
 
+    /**
+     * Export a ML model in PMML format.
+     *
+     * @param model the model to be exported
+     * @return PMML model as a String
+     * @throws MLPmmlExportException
+     */
+    public String exportAsPMML(MLModel model) throws MLPmmlExportException {
+        Externalizable extModel = model.getModel();
+
+        try {
+            if (extModel instanceof PMMLModelContainer) {
+                PMMLExportable pmmlExportableModel = ((PMMLModelContainer) extModel).getPMMLExportable();
+                String pmmlString = pmmlExportableModel.toPMML();
+
+                DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+                DocumentBuilder builder;
+
+                try {
+                    //convert the string to xml to append the version attribute
+                    builder = factory.newDocumentBuilder();
+                    Document document = builder.parse(new InputSource(new StringReader(pmmlString)));
+                    Element root = document.getDocumentElement();
+                    root.setAttribute("version", "4.2");
+
+                    // convert it back to string
+                    StringWriter stringWriter = new StringWriter();
+                    TransformerFactory tf = TransformerFactory.newInstance();
+                    Transformer transformer = tf.newTransformer();
+                    transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "no");
+                    transformer.setOutputProperty(OutputKeys.METHOD, "xml");
+                    transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+                    transformer.setOutputProperty(OutputKeys.ENCODING, "UTF-8");
+
+                    transformer.transform(new DOMSource(document), new StreamResult(stringWriter));
```

Comment:
Can you move the version appending logic to a helper method? since this is a temporary fix, please mention it in a line comment.